lib/src/rules/avoid_defining_a_class_that_contains_only_static_members.dart

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library linter.src.rules.avoid_defining_a_class_that_contains_only_static_members;
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:linter/src/analyzer.dart';
+
+const _desc = r'Avoid defining a class that contains only static members.';
+
+const _details = r'''
+
+**AVOID** defining a class that contains only static members.
+
+**BAD:**
+```
+class DateUtils {
+  static DateTime mostRecent(List<DateTime> dates) {
+    return dates.reduce((a, b) => a.isAfter(b) ? a : b);
+  }
+}
+
+class _Favorites {
+  static const mammal = 'weasel';
+}
+```
+
+**GOOD:**
+```
+DateTime mostRecent(List<DateTime> dates) {
+  return dates.reduce((a, b) => a.isAfter(b) ? a : b);
+}
+
+const _favoriteMammal = 'weasel';
+```
+
+''';
+
+bool _isNonConst(FieldElement element) => !element.isConst;
+
+bool _isStaticMember(ClassMember classMember) {
+  if (classMember is MethodDeclaration) {
+    return classMember.isStatic;
+  }
+  if (classMember is FieldDeclaration) {
+    return classMember.isStatic;
+  }
+  return false;
+}
+
+class AvoidDefiningAClassThatContainsOnlyStaticMembers extends LintRule {
+  _Visitor _visitor;
+  AvoidDefiningAClassThatContainsOnlyStaticMembers()
+      : super(
+            name: 'avoid_defining_a_class_that_contains_only_static_members',
```

Comment:
Perhaps we can find a shorter name, such as 'avoid_classes_with_only_static_members' (although something even shorter would be nice).